### PR TITLE
HW-52168 apply new fields structure to hw sdk

### PR DESCRIPTION
**Pull request by peter-joseph**

### CODE REVIEW

File: core/src/test/java/com/hyperwallet/android/model/meta/HyperwalletRetrieveTransferMethodConfigurationFieldsTest.java [hunk 1]
```diff
@@ -44,74 +43,126 @@
...
+                .get(2).getName(), is("addressLine1"));
+        assertThat(resultFields.getFields().getFieldGroups().get(5).getFields()
+                .get(3).getName(), is("city"));
+        assertThat(resultFields.getFields().getFieldGroups().get(5).getFields()
+                .get(4).getName(), is("postalCode"));
+
+        // assert fees
+        assertThat(resultFields.getFees(), Matchers.<HyperwalletFee>hasSize(1));
+        assertThat(resultFields.getFees().get(0).getValue(), is("5.00"));
+        assertThat(resultFields.getFees().get(0).getFeeRateType(), is("FLAT"));
```
Comment (vshcherbyna-epam) [inline core/src/test/java/com/hyperwallet/android/model/meta/HyperwalletRetrieveTransferMethodConfigurationFieldsTest.java]: Could we expect here the FeeRate.FLAT

File: core/src/test/java/com/hyperwallet/android/model/meta/HyperwalletRetrieveTransferMethodConfigurationFieldsTest.java [hunk 2]
```diff
@@ -44,74 +43,126 @@
...
+        verify(mListener, never()).onFailure(any(HyperwalletException.class));
+        HyperwalletTransferMethodConfigurationField resultFields = mResultArgumentCaptor.getValue();
+
+        // assert fields
+        assertThat(resultFields.getFields(), is(notNullValue()));
+        assertThat(resultFields.getFields().getFieldGroups(), is(notNullValue()));
+        assertThat(resultFields.getFields().getFieldGroups(),
+                Matchers.<HyperwalletFieldGroup>hasSize(6));
+        // check group fields sequence
+        assertThat(resultFields.getFields().getFieldGroups().get(0).getGroupName(),
```
Comment (azakrevska-epam) [inline core/src/test/java/com/hyperwallet/android/model/meta/HyperwalletRetrieveTransferMethodConfigurationFieldsTest.java]: @peter-joseph May be "resultFields.getFields().getFieldGroups()" can be extracted to variable, for example : fieldGroups = resultFields.getFields().getFieldGroups() and why we should check all items?

Comment (peter-joseph) [reply core/src/test/java/com/hyperwallet/android/model/meta/HyperwalletRetrieveTransferMethodConfigurationFieldsTest.java]: this is just to make sure that we are retaining the sequence of response from GQL our SDK is not sorting and nodes

File: core/src/main/java/com/hyperwallet/android/model/meta/field/HyperwalletFileSize.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import androidx.annotation.NonNull;
+
+import org.json.JSONObject;
+
+/**
+ * @code HyperwalletFileSize representation
+ */
+public class HyperwalletFileSize {
+
+    private static String MIN = "min";
```
Comment (fmattos-hw) [inline core/src/main/java/com/hyperwallet/android/model/meta/field/HyperwalletFileSize.java]: MIN and MAX should be final

File: core/src/main/java/com/hyperwallet/android/model/meta/query/HyperwalletTransferMethodConfigurationFieldQuery.java [hunk 4]
```diff
@@ -94,12 +100,13 @@
...
      * @param currency           the 3 letter ISO 4217-1 currency code
-     * @param transferMethodType {@code BANK_ACCOUNT}, {@code BANK_CARD}, {@code PAPER_CHECK} or {@code WIRE_ACCOUNT}
+     * @param transferMethodType @see
+     * {@link com.hyperwallet.android.model.HyperwalletTransferMethod.TransferMethodTypes}
      * @param profile            {@code INDIVIDUAL} or {@code BUSINESS}
      */
     public HyperwalletTransferMethodConfigurationFieldQuery(final @NonNull String country,
             final @NonNull String currency,
-            final @NonNull String transferMethodType,
+            final @NonNull @HyperwalletTransferMethod.TransferMethodType String transferMethodType,
```
Comment (fmattos-hw) [inline core/src/main/java/com/hyperwallet/android/model/meta/query/HyperwalletTransferMethodConfigurationFieldQuery.java]: nice! 🥇

**Pull request merged**